standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/common/metrics/metrics2/CodahaleMetrics.java

```diff
@@ -197,6 +200,10 @@ public void close() throws Exception {
     for (Map.Entry<String, Metric> metric : metricRegistry.getMetrics().entrySet()) {
       metricRegistry.remove(metric.getKey());
     }
+    if (hadoopMetricsStarted) {
+      DefaultMetricsSystem.shutdown();
```

Comment:
I guess the DefaultMetricsSystem.shutdown() is idempotent, so we can call this anytime, however, it prints an ugly exception to the logs from MetricsSystemImpl.stop(), so even if I don't like this started boolean flagging, it's the way to go